src/librustc_codegen_llvm/sir.rs

```diff
@@ -49,23 +54,29 @@ pub fn write_sir<'tcx>(tcx: TyCtxt<'tcx>, sir_llvm_module: &mut ModuleLlvm) {
 
     encoder.done().unwrap();
 
-    let (sir_llcx, sir_llmod) = (&*sir_llvm_module.llcx, sir_llvm_module.llmod());
+    let (sir_llcx, sir_llmod) = (&*llvm_module.llcx, llvm_module.llmod());
     let llmeta = common::bytes_in_context(sir_llcx, &buf);
     let llconst = common::struct_in_context(sir_llcx, &[llmeta], false);
 
     // Borrowed from exported_symbols::metadata_symbol_name().
     let sym_name = format!(
-        "{}_{}_{}",
+        "{}_{}_{}_{}_sym",
```

Comment:
Are we using this symbol name for the debug labels? If so we'll need to update the hardware tracer after this (see https://github.com/softdevteam/yk/blob/master/yktrace/src/hwt/mapper.rs#L163).

Reply:
After looking closer, I don't think we do. I guess running the tests will tell us.

Reply:
No, it uses the function symbol name.

https://github.com/softdevteam/ykrustc/blob/b86c6ecb3121f86e29a5a873a0038dfb1558d9db/src/librustc_codegen_llvm/builder.rs#L183